libgalois/src/MemoryPolicy.cpp

```diff
@@ -0,0 +1,213 @@
+#include "katana/MemoryPolicy.h"
+
+#include <fstream>
+
+#include "katana/MemoryPolicy.h"
+#include "katana/ProgressTracer.h"
+
+// anchor vtable
+katana::MemoryPolicy::~MemoryPolicy() = default;
+
+struct katana::MemoryPolicy::MemInfo {
+  count_t active;
+  count_t standby;
+  uint64_t rss_bytes;
+  double used_ratio;
+  int64_t oom_score;
+};
+
+using katana::count_t;
+
+namespace {
+
+std::string
+ToGB(uint64_t total) {
+  return fmt::format("{:.3f}", total / 1024.0 / 1024.0 / 1024.0);
+}
+
+void
+LogIt(const std::string& str, katana::MemoryPolicy::MemInfo* mem_info) {
+  auto scope = katana::GetTracer().StartActiveSpan(str);
+  scope.span().Log(
+      "mem_stats", {
+                       {"rss_gb", ToGB(mem_info->rss_bytes)},
+                       {"oom_score", mem_info->oom_score},
+                       {"used_ratio", mem_info->used_ratio},
+                       {"active_gb", ToGB(mem_info->active)},
+                       {"standby_gb", ToGB(mem_info->standby)},
+                   });
+}
+
+}  // namespace
+
+void
+katana::MemoryPolicy::UpdateMemInfo(
+    MemInfo* mem_info, count_t active, count_t standby) const {
+  mem_info->active = active;
+  mem_info->standby = standby;
+  mem_info->rss_bytes = katana::ProgressTracer::ParseProcSelfRssBytes();
+  mem_info->used_ratio =
+      (physical() == 0) ? 0.0 : (double)mem_info->rss_bytes / physical();
```

Comment:
I know this is the corner case of all corner cases, but 0.0 seems like a weird value here. Would you consider using `NaN`?

Reply:
No.

But I did ensure a non-zero value.